tests/components/Microsoft.DotNet.UpgradeAssistant.Extensions.Tests/ExtensionConfigurationTests.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Text.Json;
+using AutoFixture;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Tests
+{
+    public class ExtensionConfigurationTests
+    {
+        private readonly Fixture _fixture;
+
+        public ExtensionConfigurationTests()
+        {
+            _fixture = new Fixture();
+        }
+
+        [Fact]
+        public void ItemsSet()
+        {
+            // Arrange
+            var expected = _fixture.Create<Option1>();
+            var manifests = new[]
+            {
+                new Option1 { A = expected.A },
+                new Option1 { B = expected.B },
+            };
+
+            using var provider = CreateServiceProvider(manifests);
+
+            // Act
+            var options = provider.GetRequiredService<IOptions<Option1>>();
+
+            // Assert
+            Assert.Equal(expected, options.Value);
+        }
+
+        [Fact]
+        public void SingleArray()
+        {
+            // Arrange
+            var manifests = new[]
+            {
+                new OptionsWithArray { Array = new[] { 1, 2 } },
+            };
+
+            using var provider = CreateServiceProvider(manifests);
+
+            // Act
+            var options = provider.GetRequiredService<IOptions<OptionsWithArray>>();
+
+            // Assert
+            Assert.Equal(new[] { 1, 2 }, options.Value.Array);
+        }
+
+        [Fact]
+        public void LastArrayIsUsed()
+        {
+            // Arrange
+            var manifests = new[]
+            {
+                new OptionsWithArray { Array = new[] { 1, 2 } },
+                new OptionsWithArray { Array = new[] { 3, 4 } },
+            };
+
+            using var provider = CreateServiceProvider(manifests);
+
+            // Act
+            var options = provider.GetRequiredService<IOptions<OptionsWithArray>>();
+
+            // Assert
+            var o = options.Value;
+            Assert.Equal(new[] { 3, 4 }, options.Value.Array);
```

Comment:
We should also confirm that `GetRequiredService<IOptions<ICollection<OptionsWithArray>>` gives both arrays back.